### ISSUE TITLE
[Accessibility][Screen Reader - Dialog]: In scan mode, Narrator focus moves out from the dialog and lands on background elements when navigate back by "Up" key on "Blocking Dialog".

### ISSUE THREAD

**Issue by TestAccessibilityGIT** [OfficeDev/office-ui-fabric-react#8134]
**Test environment: ​**
Platform: Windows 10​
OS Build : Build 18334.19h1_release.190205-1505
Browser: Microsoft Edge 44.18334.1.0
Screen reader: Narrator​
 ​
**Repro steps:​**
1. Open [URL](https://developer.microsoft.com/en-us/fabric#/components).
2. Navigate to "Surfaces" button in left navigation pane and activate it.
3. Navigate to "Dialog" button and activate it.
4. Navigate to "Open Dialog" button under "Blocking Dialog".
5. Navigate the dialog with scan mode
6. Observe that Narrator focus in scan mode moves out of the dialog when navigating back with "Up" key of keyboard.

**Actual result: ​**
Narrator focus in scan mode moves out of the "All Emails Together" dialog when navigating back with "Up" key of keyboard.

**Expected result:​**
Narrator focus in scan mode should reside inside the "All Emails Together" dialog when navigating back with "Up" key of keyboard.

**Observation:**
Same issue repro with NVDA+Chrome

**Attachment:**
[Bug_8134_1](https://microsoft-my.sharepoint.com/:v:/g/personal/v-shshu_microsoft_com1/EZ-z_vd-CoNEuEFm3azUGsoBBu4Pmc7FV3N-FJQDHPsAbQ?e=eNXr0j); [Bug_8134_2](https://microsoft-my.sharepoint.com/:v:/g/personal/v-shshu_microsoft_com1/EZ-z_vd-CoNEuEFm3azUGsoBBu4Pmc7FV3N-FJQDHPsAbQ?e=eNXr0j).
@micahgodbolt 

Tags:
HCL-MSTeams; Shifts; HCL-Fabric; A11yMAS; MAS4.2.10; Web-Chrome; Web-Edge

**Comment by leddie24** [comment]
@micahgodbolt or @JasonGore , do you have any insight on this issue? I did some digging around and I'm not sure if this is an Narrator issue, or a Fabric issue.  Dialog uses `FocusTrapZone` with a `isClickableOutsideFocusTrap={false}` if it's a blocking dialog, however even testing a simple `FocusTrapZone` with that prop set to false allows the Narrator scan mode to visit outside elements.

I found a stackoverflow issue mentioning the usage of `aria-modal` but it doesn't seem widely supported at the moment.  A workaround solution that some users suggested is to add `aria-hidden` on elements that are supposed to be blocked while the Dialog is active (in our scenario, it would probably be the `<Fabric>` component wrapper, since the `Modal` is appended to the end of the DOM.  

https://stackoverflow.com/questions/54288847/aria-modal-not-supported-in-jaws-18

Any suggestions here?

**Comment by TestAccessibilityGIT** [comment]
+Amit

**Comment by JasonGore** [comment]
Does `aria-modal` properly trap even in scan mode? It might be worth testing against #8216 as well to see if behavior is the same or different with the FTZ refactor. 

Either way, scan mode functions to some extent by ignoring intended behavior and not all browsers support `aria-modal`. This may be something we have to check with the Edge/Narrator teams about as we're probably not the first to have encountered this issue.

**Comment by leddie24** [comment]
It looks like `aria-modal` is already in place on the Popup component (see line 167 in Modal.base.tsx), and it doesn't properly trap in scan mode.  I tried testing against your link you provided earlier (http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/8216/merge/fabric-website-resources/dist/index.html#/examples/datepicker) and it doesn't trap the focus in the Dialog either.

**Comment by JasonGore** [comment]
If `aria-modal` does not trap this is likely an MS Edge or Narrator bug. May be worth going through their issue repos to see if they've already captured this and enter a bug if not.

**Comment by leddie24** [comment]
Closing this issue.  This is a dupe of #6422 and #6535.  We already have the correct aria in place (aria-modal=true). There isn't much we can do aside from wait for a fix from the Narrator team to respect the aria-modal attribute.

**Comment by jamesedjac** [comment]
@leddie24 can we consider reopening this issue?

While it is true that the ARIA spec does state that aria-modal="true" should trap focus, this has been the case for a while, and I am not aware of a browser screen reader combination that has implemented it successfully.

To make sure I am up to date on support for this property, I tested it with a variety of screen readers and browsers and the only combination where I could get it to work in a consistent manner for any combination (it may work with NVDA and Chrome, but it does not appear to be consistent and I could not rule out quirks of the specific environment I was testing in).

This does not surprise me, because screen readers are often reluctant to implement or fully implement aspects of specifications that would be easy to misapply and would cause major accessibility issues if applied incorrectly. For instance, most screen readers don't implement aria-label exactly to specification, only honoring it on controls that it anticipates a legitimate need for it. In the case of aria-modal, developers who implement it without fully understanding how it is meant to be used, could easily render whole applications inaccessible, so I could understand why screen reader would be reluctant to implement it.

Historically, the method used to achieve the expected behavior of screen reader focus being trapped in the dialog was to apply aria-hidden="true" to elements outside of the dialog. This method is [even mentioned in the ARIA Authoring Practices design pattern for Modal dialogs](https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal). In most implementations of modal dialogs, there is a wrapper around the non-modal content of the page and the modal elements are added to the DOM outside of that wrapper, making it easy to toggle and aria-hidden state on that wrapper.

Would it be possible for fabric to add an optional attribute to the dialog control where users could specify elements that should be hidden while the dialog is open? That way, fabric is not adding aria-hidden attributes to elements without the developer's specifying that this should be done.

**Comment by JasonGore** [comment]
@jamesedjac I think the method you describe of applying `aria-hidden` to elements outside of the `Dialog` is actually under the control of content authors and not something that should ever be performed by `Dialog` itself.

Regarding behavior of `aria-modal`, this bug is essentially a duplicate of #6535 since `Dialog` uses `FocusTrapZone`.
